src/main/java/net/minecraftforge/client/gui/tabs/GuiTab.java

```diff
@@ -0,0 +1,318 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package net.minecraftforge.client.gui.tabs;
+
+import java.util.Arrays;
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.FontRenderer;
+import net.minecraft.client.gui.Gui;
+import net.minecraft.client.gui.inventory.GuiContainer;
+import net.minecraft.client.gui.inventory.GuiContainerCreative;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.client.renderer.InventoryEffectRenderer;
+import net.minecraft.client.renderer.RenderHelper;
+import net.minecraft.client.renderer.RenderItem;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+import com.google.common.base.Predicates;
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.ListMultimap;
+
+/**
+ * Class used to add a new GuiTab
+ */
+@SideOnly(Side.CLIENT)
+public class GuiTab
+{
+    private static ListMultimap<Class<? extends GuiContainer>, GuiTab> tabRegistry = ArrayListMultimap.create();
+
+    private ItemStack iconStack = ItemStack.EMPTY;
+    @Nullable
+    private ResourceLocation iconResLoc = null;
+    private String name;
+    protected Class<? extends GuiContainer> parentContainer;
```

Comment:
Is there any real reason why this needs to be restricted to only GuiContainers? Could it not also be used for non-container GUIs?

Reply:
Tried doing it to any containers but seems like there are a few methods/fields that are required in the GUI Container. If you could suggest a better solution/workaround would be happy to hear it too as well.

Reply:
Do you know which methods / fields were missing from GUI?

Reply:
It was a about a month ago When I first tested, dunno... Will try again tonight and will show results/errors

Reply:
Sorry, wont be able to test tonight, will try my best to test it as soon as I've got time :/

Reply:
Ah, I couldn't actually make it work, because it would need to set the postion of the tabs manually, which is something you can bypass when using GuiContainers, because they have a fixed pos for their background and textures. would be wierd if this was allowed for Non GuiContainers imo.